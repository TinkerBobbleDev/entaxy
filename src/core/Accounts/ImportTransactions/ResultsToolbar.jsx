import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import PropTypes from 'prop-types'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import Switch from '@material-ui/core/Switch'
import TableToolbar from '../../../common/TableToolbar'

const styles = {
  chekbox: {
    width: 200
  }
}

export class ResultsToolbarComponent extends React.Component {
  state = {
    showOnlyErrors: false
  }

  onChange = ({ target }) => {
    if (target.checked) {
      this.props.filterProps.setFilter({
        attr: 'errors',
        value: this.props.filterTransactionsWithErrors
      })
    } else {
      this.props.filterProps.unsetFilter({ attr: 'errors' })
    }
    this.setState({
      showOnlyErrors: target.checked
    })
  }

  render() {
    const {
      classes,
      title,
      subTitle,
      selectedTransactions
    } = this.props
    return (
      <TableToolbar
        title={title}
        subTitle={subTitle}
        selectedItems={selectedTransactions}
      >
        <FormControlLabel
          label="Show only errors"
          className={classes.chekbox}
          control={
            <Switch
              checked={this.state.showOnlyErrors}
              onChange={this.onChange}
              value="showOnlyErrors"
            />
          }
        />
      </TableToolbar>
    )
  }
}

ResultsToolbarComponent.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.node.isRequired,
  selectedTransactions: PropTypes.array.isRequired,
  filterProps: PropTypes.object.isRequired,
  filterTransactionsWithErrors: PropTypes.func.isRequired
}

export default withStyles(styles)(ResultsToolbarComponent)
