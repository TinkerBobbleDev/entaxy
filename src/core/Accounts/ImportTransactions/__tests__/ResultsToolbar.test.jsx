import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ResultsToolbar from '../ResultsToolbar'

const account = { id: 1, name: 'Checking', institution: 'TD' }
const mockHandleNextStep = jest.fn()
const mockHandlePrevStep = jest.fn()
const mochaSetFilter = jest.fn()
const mochaUnsetFilter = jest.fn()

afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('ResultsToolbar', () => {
  it('it renders correctly', () => {
    const props = {
      title: 'Title',
      subTitle: 'subTitle',
      account,
      selectedTransactions: [],
      filterProps: { filters: {} },
      handleNextStep: mockHandleNextStep,
      handlePrevStep: mockHandlePrevStep
    }
    const { getByText } = render(<ResultsToolbar {...props} />)
    getByText('Title')
    getByText('subTitle')
  })

  it('should show/hide only transactions with errors', async () => {
    const props = {
      title: 'Title',
      subTitle: 'subTitle',
      account,
      selectedTransactions: [],
      filterProps: {
        filters: {},
        setFilter: mochaSetFilter,
        unsetFilter: mochaUnsetFilter
      },
      handleNextStep: mockHandleNextStep,
      handlePrevStep: mockHandlePrevStep
    }
    const { getByTestId } = render(<ResultsToolbar {...props} />)

    const checkbox = getByTestId('showOnlyErrors')
    expect(checkbox.checked).toBe(false)

    fireEvent.click(checkbox)
    expect(mochaSetFilter).toHaveBeenCalledWith({ attr: 'errors', value: true })
    expect(checkbox.checked).toBe(true)

    fireEvent.click(checkbox)
    expect(mochaUnsetFilter).toHaveBeenCalledWith({ attr: 'errors' })
  })

  it('should handle submit', async () => {
    const props = {
      title: 'Title',
      subTitle: 'subTitle',
      account,
      selectedTransactions: [],
      filterProps: { filters: {} },
      handleNextStep: mockHandleNextStep,
      handlePrevStep: mockHandlePrevStep
    }
    const { getByTestId } = render(<ResultsToolbar {...props} />)
    fireEvent.click(getByTestId('saveButton'))
    expect(mockHandleNextStep).toHaveBeenCalled()
  })

  it('should handle back', async () => {
    const props = {
      title: 'Title',
      subTitle: 'subTitle',
      account,
      selectedTransactions: [],
      filterProps: { filters: {} },
      handleNextStep: mockHandleNextStep,
      handlePrevStep: mockHandlePrevStep
    }
    const { getByTestId } = render(<ResultsToolbar {...props} />)
    fireEvent.click(getByTestId('backButton'))
    expect(mockHandlePrevStep).toHaveBeenCalled()
  })
})
