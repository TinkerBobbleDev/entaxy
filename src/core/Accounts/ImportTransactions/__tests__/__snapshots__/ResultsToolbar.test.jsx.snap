// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsToolbar matches snapshot with no selected transactions 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular TableToolbar-root-1 MuiToolbar-gutters"
>
  <div
    className="TableToolbar-title-5"
  >
    <div>
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Title
      </h6>
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        subTitle
      </span>
    </div>
  </div>
  <div
    className="TableToolbar-spacer-3"
  />
  <div
    className="TableToolbar-actions-4"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-57 MuiSwitch-switchBase MuiSwitch-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiSwitch-thumb"
            />
            <input
              checked={false}
              className="PrivateSwitchBase-input-60 MuiSwitch-input"
              onChange={[Function]}
              type="checkbox"
              value="showOnlyErrors"
            />
          </span>
        </span>
        <span
          className="MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Show only errors
      </span>
    </label>
  </div>
</div>
`;

exports[`ResultsToolbar matches snapshot with some selected transactions 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular TableToolbar-root-1 TableToolbar-highlight-2 MuiToolbar-gutters"
>
  <div
    className="TableToolbar-title-5"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit"
    >
      2
      Â selected
    </h6>
  </div>
  <div
    className="TableToolbar-spacer-3"
  />
  <div
    className="TableToolbar-actions-4"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-57 MuiSwitch-switchBase MuiSwitch-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiSwitch-thumb"
            />
            <input
              checked={false}
              className="PrivateSwitchBase-input-60 MuiSwitch-input"
              onChange={[Function]}
              type="checkbox"
              value="showOnlyErrors"
            />
          </span>
        </span>
        <span
          className="MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Show only errors
      </span>
    </label>
  </div>
</div>
`;
